application/common/components/passwordStore/Multiple.php

```diff
@@ -88,48 +60,70 @@ public function getMeta($employeeId): UserPasswordMeta
     }
     
     /**
-     * See if all of the password stores seem to be available/responding, and if
-     * so set the user's password in all of the defined password stores. If any
-     * of the password stores fail the "pre-check", this will not attempt to set
-     * the user's password on any of them, instead throwing a
-     * NotAttemptedException.
+     * Set the user's password in all of the defined password stores. If any
+     * of the password stores fail, a PasswordStoreException will be thrown
+     * with a message detailing which ones succeeded and which ones failed.
      *
      * NOTE: If successful, this will return the UserPasswordMeta returned by
      *       the first password store defined in its list.
      *
      * @param string $employeeId The Employee ID of the user.
      * @param string $password The new password.
      * @return UserPasswordMeta
-     * @throws NotAttemptedException
+     * @throws PasswordStoreException
      * @throws Exception
      * @throws UserNotFoundException
      * @throws AccountLockedException
      * @throws PasswordReuseException
      */
     public function set($employeeId, $password): UserPasswordMeta
     {
-        $this->assertAllBackendsAreAvailable($employeeId, 'set the password');
-        $numSuccessfullySet = 0;
         $responses = [];
+        $successes = [];
+        $errors = [];
         foreach ($this->passwordStores as $passwordStore) {
             try {
                 $responses[] = $passwordStore->set($employeeId, $password);
-                $numSuccessfullySet++;
+                $successes[] = $passwordStore->getDisplayName();
             } catch (Exception $e) {
                 if ($e instanceof PasswordReuseException) {
                     // Be aware that this does not include information about how many backends the password
                     // was successfully changed in
                     throw $e;
                 }
-                throw new Exception(sprintf(
-                    'Failed to set the password using %s after successfully '
-                    . 'setting it in %s other password store(s). Error: %s',
-                    \get_class($passwordStore),
-                    $numSuccessfullySet,
-                    $e->getMessage()
-                ), 1498162884, $e);
+
+                \Yii::error([
+                    'action' => 'set password',
+                    'status' => 'error',
+                    'passwordStore' => $passwordStore->getDisplayName(),
+                    'message' => $e->getMessage(),
+                ]);
+                $errors[] = $passwordStore->getDisplayName();
+            }
+        }
+
+        if (count($errors) > 0) {
+            if (count($successes) > 0) {
+                $errorMessage = \Yii::t(
+                    'app',
+                    'Multiple.SetPartialSuccess {successes} {errors}',
```

Comment:
You may want to put a more visible delimiter between the successes and errors than just a space (especially since names can include a space), possibly with some sort of label of which list is which.

For example, `Multiple.SetPartialSuccess Google, ID Broker LDAP` doesn't seem to clearly indicate that "Google" and "ID Broker" succeeded and "LDAP" failed.

Reply:
Ohhh... this isn't actually the error message. Wow, I didn't realize that just from reading this code.

Reply:
Yeah, I had been omitting the interpolations in the translation key string, but this one choked the unit test because it was looking at the content of the errors list. If I could figure out how to initialize Yii in behat, I could probably change it to simply `'Multiple.SetPartialSuccess'`